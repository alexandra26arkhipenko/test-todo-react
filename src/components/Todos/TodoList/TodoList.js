import { useSelector } from "react-redux";

import Card from "../../UI/Card";

import classes from "./TodoList.module.css";

const TodoList = () => {
  const todos = useSelector((state) => state.todos);

  return (
    <Card className={classes.todos}>
      <ul>
        {todos.map((todo) => (
          <li key={todo.id}>{todo.text}</li>
        ))}
      </ul>
    </Card>
  );
};

export default TodoList;
