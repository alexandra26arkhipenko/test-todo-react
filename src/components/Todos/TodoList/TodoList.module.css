.todos {
  margin: 2rem auto;
  width: 90%;
  max-width: 40rem;
  font-family: Century Gothic, serif;
}

.todos ul {
  list-style: none;
  padding-inline-start: 0px;
  max-height: 255px;
  overflow-y: auto;
}

.todos ul::-webkit-scrollbar {
  width: 0.3em;
}
.todos ul::-webkit-scrollbar-track {
  box-shadow: inset 0 0 6px rgb(238, 230, 221);
  margin-top: 10px;
  margin-bottom: 10px;
}

.todos ul::-webkit-scrollbar-thumb {
  background-color: #a47373;
  outline: 1px solid rgb(238, 230, 221);
  border-radius: 8px;
}

.todos li {
  border-bottom: 1px solid #ccc;
  padding: 0.5rem;
  margin-left: 30px;
  margin-right: 30px;
  height: 50px;
  padding-top: 20px;
  padding-bottom: 10px;
  align-items: center;
  text-align: left;
  display: flex;
}

.todos li:last-child {
  border-bottom: none;
}

.todos h3 {
  font-size: 14px;
  text-align: center;
  color: #a25e5e;
  padding: 0.5rem;
  margin-left: 18%;
  height: 50px;
  padding-top: 20px;
  align-items: center;
  display: flex;
}
