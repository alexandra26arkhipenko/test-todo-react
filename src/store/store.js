import { configureStore } from "@reduxjs/toolkit";
import { localStorageMiddleware } from "./middleware/localStorageMiddleware";

import todoReducer from "./todo";

const store = configureStore({
  reducer: todoReducer,
  middleware: [localStorageMiddleware],
});

export default store;
