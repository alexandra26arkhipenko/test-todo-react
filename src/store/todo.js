import { createSlice } from "@reduxjs/toolkit";
import { v4 as uuid } from "uuid";
import { getItemFromLocalStorage } from "../repository/localStorageRepository";
import { TODO_FOOTER_FILTER_ALL } from "../constants/constants";

const savedTodos = getItemFromLocalStorage("todos");
let todos = [];

if (savedTodos) {
  todos = JSON.parse(savedTodos);
}

const initialState = {
  todos: todos,
  filter: "",
};

const todoSlice = createSlice({
  name: "counter",
  initialState,
  reducers: {
    addTodo(state, action) {
      const newTodo = {
        id: uuid(),
        text: action.payload,
        isCompleted: false,
      };

      state.todos.unshift(newTodo);
    },
    compliteTodo(state, action) {
      const index = state.todos.findIndex((todo) => todo.id === action.payload);

      if (index !== -1) {
        state.todos[index].isCompleted = !state.todos[index].isCompleted;
      }
    },
    deleteTodo(state, action) {
      const index = state.todos.findIndex((todo) => todo.id === action.payload);

      if (index !== -1) {
        state.todos.splice(index, 1);
      }
    },
    deleteAllCompletedTodos(state) {
      state.todos = state.todos.filter((todo) => !todo.isCompleted);
    },
    filterTodos(state, action) {
      state.filter = action.payload;
    },
    reorderTodos(state, action) {
      let { dragIndex, dropIndex, filteredTodos } = action.payload;

      if (state.filterValue !== TODO_FOOTER_FILTER_ALL) {
        dragIndex = state.todos.findIndex(
          (property) => property.id === filteredTodos[dragIndex].id
        );
        dropIndex = state.todos.findIndex(
          (property) => property.id === filteredTodos[dropIndex].id
        );
      }

      const updatedTodos = [...state.todos];
      const [reorderedItem] = updatedTodos.splice(dragIndex, 1);
      updatedTodos.splice(dropIndex, 0, reorderedItem);
      state.todos = updatedTodos;
    },
  },
});

export const todoActions = todoSlice.actions;
export default todoSlice.reducer;
