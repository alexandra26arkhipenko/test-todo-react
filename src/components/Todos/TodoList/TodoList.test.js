import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../../store/store";
import TodoList from "./TodoList";

describe("TodoInput", () => {
  it("should render correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <TodoList />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
