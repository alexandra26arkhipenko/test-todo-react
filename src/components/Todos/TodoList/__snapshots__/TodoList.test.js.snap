// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput should render correctly 1`] = `
<div>
  <div
    class="card todos"
  >
    <h3>
      Nothing's here :( Create new todos and get better every day!
    </h3>
    <div
      class="footer"
    >
      <span>
        Well done!
      </span>
      <div>
        <button
          class="button undefined"
          type="button"
          value="All"
        >
          All
        </button>
        <button
          class="button undefined"
          type="button"
          value="Active"
        >
          Active
        </button>
        <button
          class="button undefined"
          type="button"
          value="Completed"
        >
          Completed
        </button>
      </div>
      <button
        class="button undefined"
        type="button"
      >
        Clear Completed
      </button>
    </div>
  </div>
</div>
`;
